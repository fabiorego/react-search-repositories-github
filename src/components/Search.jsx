import React from 'react';
import '../styles/Search.css';

export default function Search(props) {
  const visibility = props.visibility;

  return (
    <div className="search" style={{ display: visibility }}>
      <div className="title-search">
        <h4>Your search will be here...</h4>
      </div>
      <lottie-player
        src="https://assets2.lottiefiles.com/temp/lf20_LJK4oD.json"
        background="transparent"
        speed="0.4"
        style={{ width: 300, height: 300 }}
        loop
        autoplay
      ></lottie-player>
    </div>
  );
}
