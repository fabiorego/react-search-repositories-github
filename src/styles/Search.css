.search {
  width: 100%;
  display: flex;
  padding-bottom: 20px;
  align-items: center;
  flex-direction: column;
  justify-content: center;
}
.title-search {
  padding-top: 20px;
  font-size: 30px;
  font-weight: 500;
  align-items: center;
  flex-direction: column;
  justify-content: center;
}
