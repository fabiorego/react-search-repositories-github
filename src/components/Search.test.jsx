import React from 'react';
import Search from './Search';
import renderer from 'react-test-renderer';

let findTextElement = function(tree, element) {
  return true;
};

test('Search snapshot', () => {
  const component = renderer.create(<Search></Search>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('Find Your search element', () => {
  let tree = renderer.create(<Search />).toJSON();
  expect(findTextElement(tree, 'Your search')).toBeDefined();
});
