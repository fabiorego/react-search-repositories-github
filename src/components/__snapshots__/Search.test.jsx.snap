// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search snapshot 1`] = `
<div
  className="search"
  style={
    Object {
      "display": undefined,
    }
  }
>
  <div
    className="title-search"
  >
    <h4>
      Your search will be here...
    </h4>
  </div>
  <lottie-player
    autoplay={true}
    background="transparent"
    loop={true}
    speed="0.4"
    src="https://assets2.lottiefiles.com/temp/lf20_LJK4oD.json"
    style={
      Object {
        "height": 300,
        "width": 300,
      }
    }
  />
</div>
`;
